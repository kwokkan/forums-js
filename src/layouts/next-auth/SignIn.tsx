import { signIn } from "next-auth/client";
import { Button, Content, Panel } from "rsuite";

export type ProviderType = Record<string, INextProvider>;

interface IProps {
    providers: ProviderType;
    callbackUrl?: string;
}

export function SignIn(props: IProps) {
    return (
        <>
            {Object.values(props.providers).map(x =>
                <Panel key={x.id} header={
                    <h3>
                        {x.name}
                    </h3>
                }>
                    <Content>
                        <Button appearance="primary" onClick={() => signIn(x.id, { callbackUrl: props.callbackUrl })}>Sign in with {x.name}</Button>
                    </Content>
                </Panel>
            )}
        </>
    );
}
