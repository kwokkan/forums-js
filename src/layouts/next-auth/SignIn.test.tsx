import { create } from "react-test-renderer";
import { ProviderType, SignIn } from "./SignIn";

test("Renders without error", () => {
    const providers: ProviderType = {
        github: {
            id: "github",
            name: "GitHub"
        }
    };

    const tree = create(
        <SignIn providers={providers} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
