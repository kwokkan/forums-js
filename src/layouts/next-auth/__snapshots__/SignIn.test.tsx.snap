// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error 1`] = `
<div
  className="rs-panel"
>
  <div
    aria-expanded={false}
    className="rs-panel-header"
    tabIndex={-1}
  >
    <h3
      className="rs-panel-title"
    >
      GitHub
    </h3>
  </div>
  <div
    className="rs-panel-body"
    role="region"
  >
    <main
      className="rs-content"
    >
      <button
        className="rs-btn rs-btn-primary"
        onClick={[Function]}
        type="button"
      >
        Sign in with 
        GitHub
        <span
          className="rs-ripple-pond"
        >
          <span
            className="rs-ripple"
          />
        </span>
      </button>
    </main>
  </div>
</div>
`;
